--- a/subprojects/gst-plugins-base/gst-libs/gst/gl/meson.build
+++ b/subprojects/gst-plugins-base/gst-libs/gst/gl/meson.build
@@ -228,7 +228,7 @@
 gl_platforms = get_option('gl_platform')
 if gl_platforms.contains('auto')
   need_platform_egl = 'auto'
-  need_platform_glx = 'auto'
+  need_platform_glx = 'no'
   need_platform_cgl = 'auto'
   need_platform_wgl = 'auto'
   need_platform_eagl = 'auto'
@@ -242,7 +242,7 @@
     if platform == 'egl'
       need_platform_egl = 'yes'
     elif platform == 'glx'
-      need_platform_glx = 'yes'
+      need_platform_glx = 'no'
     elif platform == 'cgl'
       need_platform_cgl = 'yes'
     elif platform == 'wgl'
@@ -257,7 +257,7 @@
 
 gl_winsys = get_option('gl_winsys')
 if gl_winsys.contains('auto')
-  need_win_x11 = 'auto'
+  need_win_x11 = 'no'
   need_win_wayland = 'auto'
   need_win_win32 = 'auto'
   need_win_winrt = 'auto'
@@ -284,7 +284,7 @@
   need_win_android = 'no'
   foreach winsys : gl_winsys
     if winsys == 'x11'
-      need_win_x11 = 'yes'
+      need_win_x11 = 'no'
     elif winsys == 'wayland'
       need_win_wayland = 'yes'
     elif winsys == 'win32'
