--- a/subprojects/gst-plugins-base/sys/meson.build
+++ b/subprojects/gst-plugins-base/sys/meson.build
@@ -2,8 +2,8 @@
 # FIXME: Need to check for XShmAttach inside libXext
 xshm_dep = dependency('xext', required : get_option('xshm'))
 xi_dep = dependency('xi', required : get_option('xi'))
-core_conf.set('HAVE_XSHM', xshm_dep.found())
+core_conf.set('HAVE_XSHM', false)
 core_conf.set('HAVE_XI2', xi_dep.found())
 
-subdir('ximage')
-subdir('xvimage')
+#subdir('ximage')
+#subdir('xvimage')
