--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,19 +47,19 @@
 
 IF(WIN32)
-    find_program(NUGET nuget)
-    if(NOT NUGET)
-      file(DOWNLOAD https://dist.nuget.org/win-x86-commandline/v5.6.0/nuget.exe "${CMAKE_BINARY_DIR}/nuget.exe"
-        EXPECTED_MD5 3f8a53a524c54cf2e4efb7aea370a977 STATUS status LOG log)
-      find_program(NUGET nuget HINTS "${CMAKE_BINARY_DIR}" )
-    endif()
-    message("Nuget is found at ${NUGET}")
-
-    if (CMAKE_SIZEOF_VOID_P MATCHES 8)
-      set(WIN_PLATFORM "x64")
-      set(WIN_PLATFORMx "x64")
-    else(CMAKE_SIZEOF_VOID_P MATCHES 8)
-      set(WIN_PLATFORM "win32")
-      set(WIN_PLATFORMx "x32")
-    endif(CMAKE_SIZEOF_VOID_P MATCHES 8)
+    #find_program(NUGET nuget)
+    #if(NOT NUGET)
+    #  file(DOWNLOAD https://dist.nuget.org/win-x86-commandline/v5.6.0/nuget.exe "${CMAKE_BINARY_DIR}/nuget.exe"
+    #    EXPECTED_MD5 3f8a53a524c54cf2e4efb7aea370a977 STATUS status LOG log)
+    #  find_program(NUGET nuget HINTS "${CMAKE_BINARY_DIR}" )
+    #endif()
+    message("Nuget disabled for vcpkg build")
+    set(NUGET "")
+    #if (CMAKE_SIZEOF_VOID_P MATCHES 8)
+    #  set(WIN_PLATFORM "x64")
+    #  set(WIN_PLATFORMx "x64")
+    #else(CMAKE_SIZEOF_VOID_P MATCHES 8)
+    #  set(WIN_PLATFORM "win32")
+    #  set(WIN_PLATFORMx "x32")
+    #endif(CMAKE_SIZEOF_VOID_P MATCHES 8)
 
 ENDIF()
@@ -156,7 +156,7 @@
   if(NOT HAS_GENERIC_SUPPORT)
     add_definitions("-DCOMPILER_HAS_NO_GENERIC_SUPPORT")
   else()
-    add_compile_options(/std:c11)
+    add_compile_options($<$<COMPILE_LANGUAGE:C>:/std:c11>)
   endif()
   set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
   set (CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
@@ -225,6 +225,6 @@
 endforeach()
 
 IF(WIN32)
   # This property needs to be set on a target at the root; doesn't matter which
-  set_target_properties( survive-cli PROPERTIES VS_USER_PROPS "$(SolutionDir)\\packages\\OpenBLAS.0.2.14.1\\build\\native\\openblas.targets" )
+  #set_target_properties( survive-cli PROPERTIES VS_USER_PROPS "$(SolutionDir)\\packages\\OpenBLAS.0.2.14.1\\build\\native\\openblas.targets" )
 endif()
