--- thirdparty/cmake_functions/CMakeLists.txt.orig	2023-05-16 13:27:43.881006966 -0500
+++ thirdparty/cmake_functions/CMakeLists.txt	2023-05-16 13:28:47.097147684 -0500
@@ -8,7 +8,7 @@
 # Variable definitions
 set(cmake_functions_install_cmake_dir   cmake)
 else (MSVC)
-set(cmake_functions_install_cmake_dir   lib/cmake/${PROJECT_NAME}-${PROJECT_VERSION})
+set(cmake_functions_install_cmake_dir   lib/cmake/${PROJECT_NAME})
 endif (MSVC)
 set(cmake_functions_config              ${CMAKE_CURRENT_BINARY_DIR}/cmake/${PROJECT_NAME}Config.cmake)
 set(cmake_functions_config_version      ${CMAKE_CURRENT_BINARY_DIR}/cmake/${PROJECT_NAME}ConfigVersion.cmake)
--- contrib/ecaltime/linuxptp/src/ecal_time_linuxptp.h.orig	2023-06-12 14:50:47.970371281 -0500
+++ contrib/ecaltime/linuxptp/src/ecal_time_linuxptp.h	2023-06-12 14:50:57.406428000 -0500
@@ -21,6 +21,7 @@

 #include <time.h>
 #include <mutex>
+#include <string>

 #define CLOCKFD 3
 #define FD_TO_CLOCKID(fd)   ((~(clockid_t) (fd) << 3) | CLOCKFD)
--- thirdparty/termcolor/include/termcolor/termcolor.hpp.orig	2023-06-12 15:52:40.758817294 -0500
+++ thirdparty/termcolor/include/termcolor/termcolor.hpp	2023-06-12 15:53:17.518925014 -0500
@@ -13,6 +13,7 @@
 #define TERMCOLOR_HPP_

 #include <iostream>
+#include <stdint.h>

 // Detect target's platform and set some macros in order to wrap platform
 // specific code this library depends on.
